src/lm/arpa-file-parser.h

```diff
@@ -86,9 +91,35 @@ class ArpaFileParser {
   void Read(std::istream &is, bool binary);
 
   /// Parser options.
-  const ArpaParseOptions& Options() const { return options_; }
+  const ArpaParseOptions &Options() const { return options_; }
 
  protected:
+  /// A status value returned from ConsumeNGram() overload.
+  class Status {
+   public:
+    /// Return the warninig status to print a warninig and continue.
+    static Status Warning(const std::string& message) {
+      return Status(kWarninig, message); }
```

Comment:
Warninig is misspelled.  
But actually I think I have a bigger problem with this code-- I think writing a special class for error-handling just for parsing ARPA files is just too much.  It feels like meta-programming.  Better to keep things simple.
If something is an error, then use KALDI_ERR.
if something is OK, then do nothing.
If something is a warning, then print a warning, and you can limit the number of warnings using a static namespace-scope variable declared inside the .cc file where this code lives, e.g. set it to 10.  That's how we normally rate-limit warnings.  Kaldi is distributed as source, so in the unlikely event that someone cares about the warning-limit enough to want to change it, they can modify the source.